modules/Logger/LogInstance.ts

```diff
@@ -0,0 +1,42 @@
+/*
+  Responsibilities:
+    - Provide public api
+    - Send logs to centeral place
+*/
+import type { Logger } from './Logger'
+import type { LabelKeyType } from './types'
+
+export class LogInstance {
+  private labels: string[]
+  private dispatch: Logger['dispatch']
+
+  constructor(labels: string | string[], dispatch: Logger['dispatch']) {
+    this.labels = typeof labels === 'string' ? [labels] : labels
+    this.dispatch = dispatch
+  }
+
+  // Note: filePath, methodName, lineNumber are injected at build time
+  private capture = (level: LabelKeyType) => (
+    message: string,
+    metadata?: Record<string, unknown>,
+    filePath?: string,
+    methodName?: string,
+    lineNumber?: string
```

Comment:
Some of the properties are defined differently than the contract we agreed on in the decision log.

For example;

filePath -> file
methodName -> method
lineNumber -> line

It might be worth aligning them with other platforms and the decision log.